pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -578,20 +579,25 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         """
         if self.dirpath is not None:
             return  # short circuit
-
-        if trainer.logger is not None:
+        if trainer.loggers:
             if trainer.weights_save_path != trainer.default_root_dir:
                 # the user has changed weights_save_path, it overrides anything
                 save_dir = trainer.weights_save_path
             else:
-                save_dir = trainer.logger.save_dir or trainer.default_root_dir
-
-            version = (
-                trainer.logger.version
-                if isinstance(trainer.logger.version, str)
-                else f"version_{trainer.logger.version}"
-            )
-            ckpt_path = os.path.join(save_dir, str(trainer.logger.name), version, "checkpoints")
+                if len(trainer.loggers) == 1:
+                    save_dir = trainer.logger.save_dir or trainer.default_root_dir
+                else:
+                    save_dir = trainer.default_root_dir
```

Comment:
As discussed offline, we need to be a little creative here.
Here is an trainer configuration with two loggers:
```py
trainer = Trainer(
        logger=[TensorBoardLogger("here1"), TensorBoardLogger("here2")],
        default_root_dir=os.getcwd(),
        limit_train_batches=1,
        max_epochs=2,
    )
```
Running this on master will create a directory structure like this (bug_report is the cwd here):

<img width="353" alt="image" src="https://user-images.githubusercontent.com/5495193/154774347-9a9149fa-0c0a-4421-b166-2e6f14e7c6f2.png">


On this branch, the structure will look like this:
<img width="353" alt="image" src="https://user-images.githubusercontent.com/5495193/154774508-25403b84-335d-4c20-8edc-72c2eb0f89c1.png">


The main difference here is just that the checkpoints are no longer under a version folder. This is because we want to move away from the LoggerCollection and now we have to make a hard decision where we want to save the checkpoints. We discussed options:

1. Save in default_root dir (current implementation in this PR)
2. Save to the log folder of the first logger
3. Save a copy of the checkpoint to all logger folders

We will create an RFC to discuss these options. To proceed with this PR, I propose two options:

A) Keep it as a pure refactor: We would need to introduce a utility function that produces the same path here that the LoggerCollection did for backward compatibility.
B) Make a breaking change in this PR: Go with the current implementation and just go with default_root_dir.

@daniellepintz @akashkw @carmocca We discussed B) in our meeting but I think I prefer A) for now. What are your thoughts?

Reply:
I feel like option A makes more sense to me. Changing this behavior should have its own separate discussion with its own PR. We could make a separate utility function but I don't think that will be strictly necessary. Replicating this behavior should only require 2 extra lines to concatenate the names and versions.

Reply:
@awaelchli thanks for bringing this up! Thinking about it more, I agree that option A makes the most sense! I will create a separate RFC to discuss making any changes here